src/Cardano/Wallet/Primitive/Signing.hs

```diff
@@ -0,0 +1,135 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module provides functionality for signing transactions.
+--
+-- It relies on the binary CBOR format of transactions,
+module Cardano.Wallet.Primitive.Signing
+    ( -- * Sign transactions
+      mkStdTx
+
+      -- * Helpers
+    , shuffle
+    )
+    where
+
+
+import Prelude
+
+import Cardano.Wallet.Binary
+    ( TxWitness (..), encodeTx, toByteString )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( ChangeChain (ExternalChain)
+    , Depth (AddressK, RootK)
+    , Key
+    , Passphrase
+    , SignTag (SignTx)
+    , XPrv
+    , deriveAccountPrivateKey
+    , deriveAddressPrivateKey
+    , encodeXPub
+    , publicKey
+    , sign
+    )
+import Cardano.Wallet.Primitive.AddressDiscovery
+    ( SeqState (externalPool), lookupAddress )
+import Cardano.Wallet.Primitive.Types
+    ( Address, Hash (..), Tx (..), TxIn, TxOut (..) )
+import Control.Monad
+    ( forM, forM_ )
+import Crypto.Hash
+    ( Blake2b_256, hash )
+import Crypto.Number.Generate
+    ( generateBetween )
+import Data.ByteArray
+    ( convert )
+import Data.ByteString
+    ( ByteString )
+import Data.Vector.Mutable
+    ( IOVector )
+
+import qualified Data.Vector as V
+import qualified Data.Vector.Mutable as MV
+
+-- | Build a transaction
+
+-- | Construct a standard transaction
+--
+-- " Standard " here refers to the fact that we do not deal with redemption,
+-- multisignature transactions, etc.
+--
+-- TODO: I removed FakeSigner/SafeSigner. Might be wrong.
+mkStdTx :: SeqState
+        -> (Key 'RootK XPrv, Passphrase "encryption")
+--        -> ([TxOut] -> m [TxOut])
+        -> [(TxIn, TxOut)]
+        -- ^ Selected inputs
+        -> [TxOut]
+        -- ^ Selected outputs (including change)
+        -> Maybe (Tx, [TxWitness])
```

Comment:
Careful about indentation here :pray:

Reply:
Also, perhaps, instead of a `Maybe`, have a proper signing error here? I guess the only problem could be that we cannot find the given address in the state? So, we can better report this I suppose and make it more explicit.

Reply:
Done:
```haskell
-- Signing.hs
newtype SignTxError = KeyNotFoundForAddress Address

-- Wallet.hs
data ErrSignTx
    = ErrSignTxNoSuchWallet ErrNoSuchWallet
    | ErrSignTx SignTxError